src/content/docs/ja/guides/deploy/aws.mdx

```diff
@@ -0,0 +1,258 @@
+    :::
+
+5. Bucketの Webサイトホスティングを有効にします。この設定は、バケットの**プロパティ > 静的ウェブサイトホスティング**にあります。インデックスドキュメントを `index.html` に、エラードキュメントを `404.html` に設定します。最後に、バケットの**プロパティ > 静的ウェブサイトホスティング**に、新しいウェブサイトの URL を設定します。
+
+    :::note
+    シングルページアプリケーション (SPA) をデプロイする場合は、エラードキュメントを `index.html` に設定してください。
+    :::
+
+## S3とCloudFront
+
+CloudFrontは、コンテンツ・デリバリー・ネットワーク（CDN）機能を提供するウェブサービスです。ウェブサーバーのコンテンツをキャッシュし、エンドユーザーに配信するために使用されます。CloudFrontは転送されたデータ量に応じて課金されます。CloudFrontをS3バケットに追加することで、より費用対効果が高く、より高速な配信が可能になります。
+
+CloudFrontを使用してS3バケットをラップし、AmazonグローバルCDNネットワークを使用してプロジェクトのファイルを提供します。これにより、プロジェクトのファイルを提供するコストが削減され、サイトのパフォーマンスが向上します。
+
+### S3のセットアップ
+
+1. プロジェクト名でS3バケットを作成します。
+
+    :::tip
+    バケット名はグローバルに一意でなければなりません。プロジェクト名とサイトのドメイン名の組み合わせをお勧めします。
+    :::
+2. `dist` にあるビルドしたファイルをS3にアップロードします。これは、コンソールで手動で行うか、AWS CLIを使用して行うことができます。AWS CLIを使用する場合は、[AWS認証情報で認証](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html)した後に、以下のコマンドを使用します。
+
+    ```
+    aws s3 cp dist/ s3://<BUCKET_NAME>/ --recursive
+    ```
+
+3. バケットポリシーを更新し、**CloudFront Access**を許可します。この設定は、バケットの**Permissions > Bucket policy**にあります。
+
+    ```json
+    {
+      "Version": "2012-10-17",
+      "Statement": [{
+        "Effect": "Allow",
+        "Principal": {
+          "AWS": "arn:aws:iam::cloudfront:user/CloudFront Origin Access Identity <CLOUDFRONT_OAI_ID>"
+        },
+        "Action": "s3:GetObject",
+        "Resource": "arn:aws:s3:::astro-aws/*"
+      }]
+    }
+    ```
+
+    :::caution
+    `<CLOUDFRONT_OAI_ID>` をCloudFront Origin Access Identity IDの名前に置き換えることを忘れないでください。CloudFrontのオリジンアクセスアイデンティティIDは、CloudFrontの設定後に**CloudFront > Origin access identities**で確認できます。
+    :::
+
+### CloudFrontのセットアップ
+
+1. 以下の値でCloudFrontディストリビューションを作成します：
+    * **オリジンドメイン：** 作成したS3バケット
+    * **S3バケットへのアクセス：** "OAIを使用する（バケットはCloudFrontのみにアクセスを制限できる）"
+    * **オリジンアクセスアイデンティティ：** 新しいOAIを作成
+    * **ビューワーバケットポリシー：** "いいえ、バケット・ポリシーを更新します。"
+    * **ビューワープロトコルポリシー：** "Redirect to HTTPS"
+    * **デフォルトルートオブジェクト：** `index.html`
+
+この設定は、パブリックインターネットからのS3バケットへのアクセスをブロックし、グローバルCDNネットワークを使用してサイトを提供します。CloudFrontの配信URLは、バケットの**Distributions > Domain name**で確認できます。
+
+### CloudFront Functions setup
+
+残念ながら、CloudFrontはデフォルトでは複数ページの `sub-folder/index` ルーティングをサポートしていません。これを設定するには、CloudFront Functionsを使ってリクエストをS3の目的のオブジェクトに向ける必要があります。
+
+1. 以下のコードスニペットで新しいCloudFront関数を作成します。CloudFront関数は**CloudFront > 関数**にあります。
+
+    ```js
+    function handler(event) {
+      var request = event.request;
+      var uri = request.uri;
+
+      // URIにファイル名がないかチェックする。
+      if (uri.endsWith('/')) {
+        request.uri += 'index.html';
+      }
+      // URIにファイル拡張子がないかチェックする。
+      else if (!uri.includes('.')) {
+        request.uri += '/index.html';
+      }
+
+      return request;
+    }
+    ```
+2. CloudFrontディストリビューションに関数をアタッチします。このオプションは、CloudFrontディストリビューションの**設定 > ビヘイビア > 編集 > 関数**関連で見つけることができます。
+    * **ビューワーリクエスト - 関数タイプ：** CloudFrontの関数。
+    * **ビューワーリクエスト - 関数 ARN：** 前のステップで作成した機能を選択します。
+
```

Comment:
https://github.com/withastro/docs/pull/5511 was merged after this PR was created.

```suggestion

### CloudFrontのエラーページのセットアップ

S3はデフォルトで、ファイルが見つからなかった場合には404エラーを、ファイルがプライベートであった場合には403エラーを返します。いずれの場合でも、ユーザーは汚いXMLエラーページを見ることとなります。

これを変更するには、CloudFrontディストリビューションの**設定 > エラーページ**に**カスタムエラーレスポンス**を追加します。

1. 404エラーのためのカスタムエラーレスポンスを以下の値で設定します:
    * **HTTPエラーコード:** 404: Not Found
    * **エラーレスポンスをカスタマイズ:** Yes
    * **レスポンスページのパス:** `/index.html`
    * **HTTPレスポンスコード:** 200: OK

2. 403エラーのためのカスタムエラーレスポンスを以下の値で設定します:
    * **HTTPエラーコード:** 403: Forbidden
    * **エラーレスポンスをカスタマイズ:** Yes
    * **レスポンスページのパス:** `/index.html`
    * **HTTPレスポンスコード:** 200: OK

  :::tip
  ベストなユーザー体験のために、プロジェクトに[カスタム404エラーページを作成](/ja/core-concepts/astro-pages/#カスタム404エラーページ)し、**レスポンスページのパス**を`/404.html`にセットすることもできます。
  :::

```